flytepropeller/pkg/controller/garbage_collector.go

```diff
@@ -44,19 +44,30 @@ func (g *GarbageCollector) deleteWorkflows(ctx context.Context) error {
 		s.MatchExpressions = append(s.MatchExpressions, g.labelSelectorRequirements...)
 	}
 
-	// Delete doesn't support 'all' namespaces. Let's fetch namespaces and loop over each.
-	if g.namespace == "" || strings.ToLower(g.namespace) == "all" || strings.ToLower(g.namespace) == "all-namespaces" {
+	// Delete doesn't support 'all' namespaces and comma-separated namespaces. Let's fetch namespaces and loop over each.
+	if g.namespace == "" || strings.ToLower(g.namespace) == "all" || strings.ToLower(g.namespace) == "all-namespaces" || strings.Contains(g.namespace, ",") {
 		namespaceList, err := g.namespaceClient.List(ctx, v1.ListOptions{})
 		if err != nil {
 			return err
 		}
-		for _, n := range namespaceList.Items {
-			namespaceCtx := contextutils.WithNamespace(ctx, n.GetName())
-			logger.Infof(namespaceCtx, "Triggering Workflow delete for namespace: [%s]", n.GetName())
 
-			if err := g.deleteWorkflowsForNamespace(ctx, n.GetName(), s); err != nil {
+		var namespaces []string
+		if strings.Contains(g.namespace, ",") {
+			namespaces = strings.Split(g.namespace, ",")
+		} else {
+			namespaces = make([]string, 0)
+			for _, n := range namespaceList.Items {
+				namespaces = append(namespaces, n.GetName())
+			}
+		}
```

Comment:
Treat `all` and comma-separated namespaces as the same case. Transform them into a slice of string and then loop over it.